dev/src/index.ts

```diff
@@ -1285,11 +1286,11 @@ export class Firestore implements firestore.Firestore {
    * });
    * ```
    */
-  getAll<T>(
+  getAll(
```

Comment:
After seeing that awful `this as unknown as DocumentReference` kludge in `DocumentReference.getAll()` I realized that this internal implementation signature was different from the one in `firestore.d.ts`.

So I think we should get rid of the kludge and change this to

```ts
getAll<AppModelType, DbModelType extends DocumentData>(
  ...documentRefsOrReadOptions: Array<
    firestore.DocumentReference<AppModelType, DbModelType> | firestore.ReadOptions
  >
): Promise<Array<DocumentSnapshot<AppModelType, DbModelType>>> {
```

Since the definition in `firestore.d.ts` pins the type parameters to `DocumentData` and the internal call to `getAll()` pins the type parameters to its types, there is no need for `getAll()` to support heterogenous type parameters. So we can be consistent with the "old" signature and take both type parameters.